### DISCUSSION TITLE
Forward Programing for Spektrum Rxs

### DISCUSSION THREAD

**Discussion by Mikey1968** [EdgeTX/edgetx#1785] [Q&A]
So I know Opentx has made a forward programing lua but seems like they never finished it, would you or someone make a working forward programing for spektrum rx users

**Comment by raphaelcoeffic** [comment]
> So I know Opentx has made a forward programing lua but seems like they never finished it, would you or someone make a working forward programing for spektrum rx users

Wasn't it done by the MPM guys? Frankly, I don't even know. But yes, why not, would if nice if someone would update it.

**Comment by pfeerick** [comment]
Forward programming is done via the Lua Script provided as part of the MPM - https://github.com/pascallanger/DIY-Multiprotocol-TX-Module/tree/master/Lua_scripts

Specifically, download the `DSM FwdPrg.lua` to the `/SCRIPTS/TOOLS` directory of your transmitter's SD card, and it will be visible in the SYS -> Tool screen. As well as any of the other scripts you add. 

To what extent it works, I have no idea - I don't have any receivers that use it. If you tell me a cheap one, I might even look at it (no promises though! 😛).

**Comment by Mikey1968** [comment]
Where do you download it at

**Comment by pfeerick** [comment]
Um... At the link I gave? It's the first file in the list... click on the file listing, then you can right click on the 'Raw' button over on the top right of the file view, and download the script ready to put on the SD card.

**Comment by Mikey1968** [comment]
I don't see no raw button. I clicked on the first file but see no where to
download it

**Comment by Mikey1968** [comment]
I see view RAW but there's no download

**Comment by pfeerick** [comment]
Right click on 'Raw', and choose "save link as" or equivalent for your web browser.

**Comment by pfeerick** [comment]
![image](https://user-images.githubusercontent.com/5500713/128115948-8548636d-2758-447f-8a47-de807d65c7c8.png)

**Comment by pfeerick** [comment]
@Mikey1968 How did you go with this... did you get it working?

**Comment by Mikey1968** [comment]
I haven't tried it yet. Need to buy a Rx to try it .
